sonar-db/src/main/java/org/sonar/db/ce/CeQueueDao.java

```diff
@@ -107,7 +109,7 @@ public int countAll(DbSession dbSession) {
   }
 
   public Optional<CeQueueDto> peek(DbSession session) {
-    List<String> taskUuids = mapper(session).selectEligibleForPeek();
+    List<String> taskUuids = mapper(session).selectEligibleForPeek(ONE_ROW_LIMIT);
```

Comment:
Using MyBatis RowBounds is not the perfect solution as efficiency depends on driver implementation. In a perfect world the SELECT request should define the LIMIT and OFFSET, but it's not trivial as syntax differs according to db vendor.
Still RowBounds can be considered as good enough here.